src/libpriv/rpmostree-postprocess.cxx

```diff
@@ -867,6 +868,14 @@ rpmostree_compose_commit (int            rootfs_fd,
                                  cancellable, error))
     return glnx_prefix_error (error, "While writing commit");
 
+  if (src_detached_metadata != NULL)
+    {
+        g_autoptr(GVariantDict) detached_metadata_dict = g_variant_dict_new (src_detached_metadata);
+        g_autoptr(GVariant) detached_metadata = g_variant_dict_end(detached_metadata_dict);
```

Comment:
This doesn't seem necessary. We should be able to pass in the `GVariant` as is.

Reply:
Ok, I've dropped that.